File: 6-kyu/linked-lists---length-&-count/python/solution.py
class Node(object):
    def __init__(self, data):
        self.data = data
        self.next = None
    
def length(node):
    if not node:
        return 0
    return 1 + length(node.next)

def count(node, data):
    ret = 0
    if not node:
        return ret
    if node.data == data:
        ret = 1
    return ret + count(node.next, data)

File: 6-kyu/linked-lists---length-&-count/swift/solution.swift
class Node {
    var data: Int
    var next: Node?
    init(_ data: Int) {
        self.data = data
    }
}

func length(_ head: Node?) -> Int {
    guard let node = head else { return 0 }
    return 1 + length(node.next)
}

func count(_ head: Node?, _ data: Int) -> Int {
    var ret: Int = 0
    guard let node = head else { return ret }
    if node.data == data {
        ret = 1
    }
    return ret + count(node.next, data)
}